Builds/Test.py

```diff
@@ -0,0 +1,113 @@
+#    purpose  with  or without fee is hereby granted, provided that the above
+#    copyright notice and this permission notice appear in all copies.
+#
+#    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+#    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+#    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+#    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+#    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+#    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+#    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+
+"""
+Invocation:
+  ./Builds/Test.py - builds and tests all configurations
+
+#
+# The build must succeed without shell aliases for this to work.
+#
+# Common problems:
+# 1) Boost not found. Solution: export BOOST_ROOT=[path to boost folder]
+# 2) OpenSSL not found. Solution: export OPENSSL_ROOT=[path to OpenSSL folder]
+# 3) scons is an alias. Solution: Create a script named "scons" somewhere in
+#    your $PATH (eg. ~/bin/scons will often work).
+#       #!/bin/sh
+#       python /C/Python27/Scripts/scons.py "${@}"
+"""
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import argparse
+import os
+import re
+import subprocess
+import sys
+
+BINARY_RE = re.compile(r'build/([^/]+)/rippled')
+TEST_SUCCESS_MATCH = '0 failures'
+ALL_TARGETS = 'debug', 'release', 'all'
+
+parser = argparse.ArgumentParser(
+    description='Test.py - run ripple tests'
+)
+
+parser.add_argument(
+    '--verbose', '-v',
+    action='store_true',
+    help=('Report more information about which commands are executed and the '
+          'results.'),
+)
+
+parser.add_argument(
+    'scons_args',
+    default=(),
+    nargs='*'
+    )
+
+ARGS = parser.parse_args()
+
+def shell(*cmd, **kwds):
+    "Execute a shell command and return the output."
+    verbose = kwds.pop('verbose', ARGS.verbose)
+    if verbose:
+        print('$', ' '.join(cmd))
+    processor = subprocess.call if verbose else subprocess.check_output
+    try:
+        return processor(cmd, **kwds)
+    except subprocess.CalledProcessError as e:
+        # Pretty up the exception a little with what you were doing.
+        raise ValueError('Couldn\'t execute "%s", errorcode=%s:\n%s' %
+                         (' '.join(cmd), e.returncode, e.output))
+
+if __name__ == '__main__':
+    args = ARGS.scons_args
+    if 'all' in args:
+        args = ALL_TARGETS
+    else:
+        print('Building:', *args)
+
+    # Build everything;  fail if this fails.
+    shell('scons', *args)
+
+    # Now extract the executable names and corresponding targets.
+    failed = []
+    lines = shell('scons', '-n', '--tree=derived', *args, verbose=False)
+    for line in lines.splitlines():
+        match = BINARY_RE.match(line)
+        if match:
+            executable, target = match.group(0, 1)
```

Comment:
Hah, I actually had that originally as a small function, but I inlined everything (because I tend to write a lot of functions and then inline everything that ends up being "a couple of lines" at the end).

Indeed, you don't need a function - you could just say:

`BINARY_MATCH = re.compile(r'build/([^/]+)/rippled').match`